### ISSUE TITLE
oiGet(oi,'photons noise') -- what does it do, conceptually

### ISSUE THREAD

**Issue by DavidBrainard** [isetbio/isetbio#65]
I think that when I do an oiGet(oi,'photons') what I get is the irradiance in units of photons/sec?  Is that correct?

There is also the option of doing an oiGet(oi,'photons noise').  This appears to take the irradiance values and return a Poisson draw with that as a mean.  So, I guess this would be the noisy irradiance image you'd get for a one second integration time?  

If I am thinking about things correctly, this seems like a slightly worrisome concept, since it's not the case that I can take what I get back, multiply by some other integration time and get a noisy draw of what would happen for the given retinal irradiance and integration time.

Maybe I'm missing the concept here?

**Comment by DavidBrainard** [comment]
Above, I should have written irradiance in units of photons/[sec-m2].

Also, I think what comes back from oiGet(oi,'photons noise') if I'm reading the code correctly would be the noisy image acquired in one second by a perfectly efficient sensor with a 1 m2 integration area.  That does not seem too likely to be a quantity we are interested in.

**Comment by wandell** [comment]
Yes, the oi photons are q/s/m2 for irradiance at the retina.

Concerning the oiPhotonNoise  I spent some time looking around to see where it has been used, and I can’t find any places where it is called within the main routine.

I also searched on the strings ‘photon noise’ and ‘photonnoise’ as well.

My recollection, however, is something like this.  HJ was doing some tests on classifiers and he wanted to know whether there was enough information in the OI photons to make some discrimination.  So, he wanted to make a noisy version of the oi data to run through his classifier.

This seems like something worth doing from time to time.  And the routine doesn’t seem like it is basically wrong or doing any harm.  Maybe we just need to protect ourselves from damage by doing a better job commenting?


On Jul 28, 2015, at 10:47 PM, David Brainard <notifications@github.com<mailto:notifications@github.com>> wrote:


Above, I should have written irradiance in units of photons/[sec-m2].

Also, I think what comes back from oiGet(oi,'photons noise') if I'm reading the code correctly would be the noisy image acquired in one second by a perfectly efficient sensor with a 1 m2 integration area. That does not seem too likely to be a quantity we are interested in.

**Comment by DavidBrainard** [comment]
I changed this routine so that it returns the noisy irradiance you would get if you used a collection area of one pixel of the optical image for each pixel, and for an integration time of 50 msec, and updated the comments to say that this is what it does.

I don't think returning the noisy image at each pixel for a collection area of a square meter (which is what it was doing) gives numbers that have useful meaning, while showing the noisiness at the pixel collection area does make sense to me (and indeed is what  I was writing code to do, to illustrate photon noise effects).  The choice of integration time is a little more arbitrary, but 50 msec is my rough and ready number for high photopic vision.

A person with more time could add arguments to allow integration time and collection area to be passed as additional arguments for this get, but I only added a comment that this might be a nice future enhancement.

Closing for now.  Reopen if you object.
